File: src/main/java/net/ddns/rarnold/randomworkoutroutine/controller/ExerciseController.java
package net.ddns.rarnold.randomworkoutroutine.controller;

import lombok.RequiredArgsConstructor;
import net.ddns.rarnold.randomworkoutroutine.model.Exercise;
import net.ddns.rarnold.randomworkoutroutine.service.ExerciseService;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/exercise")
@RequiredArgsConstructor
public class ExerciseController {

    private final ExerciseService exerciseService;

    @GetMapping("/next")
    public Exercise getNext() {
        return exerciseService.getRandomExercise();
    }
}

File: web/src/main/kotlin/uk/co/rafearnold/randomworkoutroutine/web/controller/ExerciseController.kt
package uk.co.rafearnold.randomworkoutroutine.web.controller

import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import uk.co.rafearnold.randomworkoutroutine.web.model.entity.Exercise
import uk.co.rafearnold.randomworkoutroutine.web.service.ExerciseService

@RestController
@RequestMapping("/api/exercise")
class ExerciseController(service: ExerciseService) : ItemController<Exercise>(service)
